python/sglang/srt/managers/schedule_batch.py

```diff
@@ -350,16 +352,22 @@ def check_finished(self):
             self.finished_reason = FINISH_MATCHED_TOKEN(matched=last_token_id)
             return
 
-        # Check stop strings
-        if len(self.sampling_params.stop_strs) > 0:
-            tail_str = self.tokenizer.decode(
-                self.output_ids[-(self.sampling_params.stop_str_max_len + 1) :]
-            )
+        if (
+            len(self.sampling_params.stop_strs) > 0
+            or len(self.sampling_params.stop_regex_strs) > 0
+        ):
+            self.stop_check_text += self.tokenizer.decode(last_token_id)
```

Comment:
You cannot decode text token by token and concatenate the string. This will lead to wrong outputs.

Reply:
Oh right, I didn't know there are tokenizers where this doesn't work. Is it then ok to decode the full text each time? Or decode a fixed window and accept that it's not 100% reliable?